arangod/Aql/OptimizerRulesFeature.cpp

```diff
@@ -474,6 +474,14 @@ void OptimizerRulesFeature::addRules() {
   registerRule("splice-subqueries", spliceSubqueriesRule, OptimizerRule::spliceSubqueriesRule,
                OptimizerRule::makeFlags());
 
+  // allow nodes to asynchronously prefetch the next batch while processing the current batch.
+  // this effectively allows parts of the query to run in parallel, but as some internal details
+  // are currently not guaranteed to be thread safe (e.g., TransactionState), this is currently
+  // disabled, and should only be activated for experimental usage at one's own risk.
+  registerRule("asynch-prefetch", asynchPrefetchRule, OptimizerRule::asynchPrefetch,
```

Comment:
Let's rename this to "async-" (not "asynch-"). I think "async" is more common than "asynch", although both would be understood.